### PR TITLE
cloudstream: fix defer cleanup

### PR DESCRIPTION
**Pull request by Iceber**
Signed-off-by: Iceber Gu <wei.cai-nat@daocloud.io>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:
`err` is overwritten. 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/cloudstream/streamserver.go [hunk 1]
```diff
@@ -140,7 +139,7 @@ func (s *StreamServer) getContainerLogs(r *restful.Request, w *restful.Response)
 		klog.Infof("Delete %s from %s", logConnection.String(), session.String())
 	}()
 
-	if err := logConnection.Serve(); err != nil {
+	if err = logConnection.Serve(); err != nil {
 		err = fmt.Errorf("apiconnection Serve %s in %s error %v",
```
Comment (fisherxu) [inline cloud/pkg/cloudstream/streamserver.go:143]: If we don't remove `:` above, the err returned maybe nil, right?

Comment (Iceber) [reply cloud/pkg/cloudstream/streamserver.go:143]: yes

Comment (fisherxu) [reply cloud/pkg/cloudstream/streamserver.go:143]: But the err here has any usage?

Comment (Iceber) [reply cloud/pkg/cloudstream/streamserver.go:143]: `WriteHeader` in the defer func
https://github.com/kubeedge/kubeedge/blob/bf51f6d2db9d83559b3f1da79e5f724492ddc433/cloud/pkg/cloudstream/streamserver.go#L100-L109

Comment (fisherxu) [reply cloud/pkg/cloudstream/streamserver.go:143]: Oh, lgtm

**Pull request merged**